lib/utils/compose_ts.ts

```diff
@@ -31,7 +32,34 @@ export interface RegistrySecrets {
 	};
 }
 
-export async function parseRegistrySecrets(
+export async function getRegistrySecrets(
+	sdk: BalenaSDK,
+	inputFilename?: string,
+): Promise<RegistrySecrets> {
+	const { fs } = await import('mz');
+	const Path = await import('path');
+
+	if (inputFilename != null) {
+		return await parseRegistrySecrets(inputFilename);
+	}
+
+	const directory = await sdk.settings.get('dataDirectory');
+	const potentialPaths = [
+		Path.join(directory, 'secrets.yml'),
+		Path.join(directory, 'secrets.yaml'),
+		Path.join(directory, 'secrets.json'),
+	];
+
+	for (const path of potentialPaths) {
+		if (await fs.exists(path)) {
```

Comment:
NIT: It would be safer to `parseRegistry` secrets without checking for `fs.exists` and catch `ENOENT`.